test/es6/default.js

```diff
@@ -284,7 +284,7 @@ var tests = [
         assert.throws(function () { eval("function f(a, b = function () { function f() { eval('1'); } }) { }") }, SyntaxError, "Evals in nested child functions are not allowed in the parameter scope", "'eval' is not allowed in the default initializer");
         assert.throws(function () { eval("function f(a, b = eval('a')) { }") }, SyntaxError, "Eval is not allowed in the parameter scope", "'eval' is not allowed in the default initializer");
         assert.throws(function () { eval("async function f(a = eval('b')) { }"); }, SyntaxError, "Eval is not allowed in the param scope of async functions", "'eval' is not allowed in the default initializer");
-        assert.throws(function () { eval("function f(a = async function(y) { eval('b'); }) { }"); }, SyntaxError, "Eval is not allowed in the param scope of nested async functions", "'eval' is not allowed in the default initializer");

```

Comment:
Unrelated to this change but I just noticed the default param eval error message

`'eval' is not allowed in the default initializer`

This is not specific enough.  Which default initializer? Is there only one?

`'eval' is not allowed in a default initializer` or
`'eval' is not allowed in the default initializer of a parameter`